blockwork/transforms/transform.py

```diff
@@ -981,19 +986,33 @@ def _input_hash(self) -> str:
         object.__setattr__(self, "_cached_input_hash", digest)
         return digest
 
-    def run(self, ctx: "Context"):
-        """Run the transform in a container."""
-        # For now serialize the instance here...
-        # later may want to make this a classmethod and pass in
-        Transform.run_serialized(ctx, self.serialize())
-
     @staticmethod
-    def run_serialized(ctx: "Context", spec: TSerialTransform):
-        """Run a transform from a spec object"""
+    def deserialize(spec: TSerialTransform) -> "Transform":
         # Get transform module
         mod = importlib.import_module(spec["mod"])
         # Get class from module (using reduce to navigate module namespacing)
         cls: Transform = reduce(getattr, spec["name"].split("."), mod)
+        tf = cls.__new__(cls)
+
+        object.__setattr__(tf, "_serial_interfaces", {})
+        for field in fields(cast(Any, tf)):
+            if not isinstance(field.default, IField):
+                if field.name == "tools":
+                    continue
+                raise ValueError(
+                    "All transform interfaces must be specified with a direction"
+                    ", e.g. `myinput: Path = Transform.IN()`"
+                )
+            ifield = field.default
+            tf._serial_interfaces[field.name] = (
+                ifield.direction,
+                SerialInterface(spec["ifaces"][field.name]),
+            )
+        return tf
+
+    def run(self, ctx: "Context") -> TSerialTransformResult:
+        """Run the transform in a container."""
+        start = time.time()
```

Comment:
Not using `datetime.now()`? I've found the datetime timedelta objects quite nice to work with in the past

Reply:
I want to keep the result object as simple jsonable data so just want a float here - in which case this is quite a bit simpler than using datetime